tests/metrics_test.go

```diff
@@ -0,0 +1,263 @@
+package tests
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	"github.com/prometheus/client_golang/api"
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/suite"
+	"io"
+	v12 "k8s.io/api/core/v1"
+	. "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+const dashboardsDir = "../operators/repository/spark/docs/latest/resources/dashboards"
+const jobName = "mock-task-runner"
+const jobTemplate = "spark-mock-task-runner-with-monitoring.yaml"
+const prometheusNamespace = "kubeaddons"
+const prometheusPod = "prometheus-prometheus-kubeaddons-prom-prometheus-0"
```

Comment:
Is there a safer way to discover Prometheus pod name instead of hardcoding the pod name? It is unlikely but quite possible that it can change after Prometheus Operator upgrade.

Reply:
good point, will try to obtain pod name at runtime via label selector